### ISSUE TITLE
SC2329 - false positive when passing function to another function

### ISSUE THREAD

**Issue by mkp76** [koalaman/shellcheck#3363]
#### For bugs with existing features

- Rule Id: SC2329
- My shellcheck version: v0.11.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

function foo1() {
	local fn=${1}
	echo "foo1"
	${fn}
}

function foo2() {
	echo "foo2"
	return 0
}

foo1 foo2

exit  0
```

#### Here's what shellcheck currently says:
Line 9:
function foo2() {
^-- [SC2329](https://www.shellcheck.net/wiki/SC2329) (info): This function is never invoked. Check usage (or ignored if invoked indirectly).

#### Here's what I wanted or expected to see:
No issue
